Neos.ContentRepository.Core/Classes/Factory/ContentRepositoryFactory.php

```diff
@@ -87,19 +120,21 @@ public function __construct(
     public function getOrBuild(): ContentRepository
     {
         if (!$this->contentRepository) {
+            $projectionStates = [];
+            foreach ($this->projectionsAndCatchUpHooks->projections as $projection) {
+                $projectionStates[] = $projection->getState();
+            }
```

Comment:
we need to exclude the main projection state here ... and maybe a projection should not return its state via `->getState();`?

Reply:
Why should the main projection state be excluded?
And what could be an alternative to getting the state off of a projection?

Reply:
> Why should the main projection state be excluded?

because users should use the security wrapped on the content repository and not access the crreadmodelinterface directly?

> And what could be an alternative to getting the state off of a projection?

if the `ProjectionFactoryInterface` has a `build` AND `buildProjectionState` or returns a touple of both on `build` ... but that are solutions for non existing problems :D Its actually simpler to have the projection know its own state, so we can flush caches on `markStale`